common/app/model/content.scala

```diff
@@ -88,11 +88,9 @@ final case class Content(
   lazy val campaigns: List[Campaign] =
     _root_.commercial.targeting.CampaignAgent.getCampaignsForTags(tags.tags.map(_.id))
 
-  lazy val isAmpSupportedArticleType: Boolean = (tags.isArticle && !tags.isLiveBlog) && !isImmersive && !tags.isQuiz
-
   lazy val shouldAmplify: Boolean = {
```

Comment:
Overall I still find this logic a bit weird, we produce a `shouldAmplifyArticles` and a `shouldAmplifyLiveBlogs` even though we already know the content type.
 
I feel this would read a lot better:
```scala
val hasBodyBlocks: Boolean = fields.blocks.exists(b => b.body.nonEmpty)
val shouldAmplifyContent = 
    if (tags.isArticle) {
         AmpArticleSwitch.isSwitchedOn && hasBodyBlocks && !isImmersive && !tags.isQuiz
    } else if (tags.isLiveBlog) {
          AmpLiveBlogSwitch.isSwitchedOn
    } else {
         false
    }
val containsFormStacks: Boolean = fields.body.contains("guardiannewsandmedia.formstack.com")
shouldAmplifyContent && !containsFormStacks
```

I think this introduces an edge case I haven't checked, where tags.isArticle & tags.isLiveBlog is true - which I don't know if is possible.

Reply:
I _think_ it can happen that `tags.isArticle` & `tags.isLiveBlog` are both true. 
[`isArticle`](https://github.com/guardian/frontend/blob/main/common/app/model/meta.scala#L820) and [`isLiveBlog`](https://github.com/guardian/frontend/blob/main/common/app/model/meta.scala#L802) are checking different tags. 

I do not 100% understand the logic but ran the debugger with this one: https://www.theguardian.com/politics/live/2022/sep/05/liz-truss-pm-prime-minister-rishi-sunak-energy-bill-freeze-tory-conservative-leader-uk-politics-live-latest-updates

and getting:
![image](https://user-images.githubusercontent.com/19683595/188485476-bb1f63dc-763f-4ce1-b807-f3bd690e07ea.png)

![image](https://user-images.githubusercontent.com/19683595/188485535-efc3ce25-7d80-4838-9ff5-5ce6428f5a5f.png)

Having said that, I totally agree with Olly that if tests are added this can be refactored and read better. Just thought I'd flag this up!

Reply:
I’ve reworked the logic to first check whether an article is a Liveblog, then if it has the blocks.

I think this should cover all use cases.